pkg/liqo-controller-manager/virtualNode-controller/namespaceMap_creation.go

```diff
@@ -0,0 +1,55 @@
+package virtualnodectrl
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutils "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+)
+
+// create a new NamespaceMap with OwnerReference.
+func (r *VirtualNodeReconciler) createNamespaceMap(ctx context.Context, n *corev1.Node) error {
+	nm := &mapsv1alpha1.NamespaceMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", n.GetAnnotations()[liqoconst.RemoteClusterID]),
+			Namespace:    liqoconst.TechnicalNamespace,
+			Labels: map[string]string{
+				liqoconst.RemoteClusterID: n.GetAnnotations()[liqoconst.RemoteClusterID],
+			},
+		},
+	}
+
+	if err := ctrlutils.SetControllerReference(n, nm, r.Scheme); err != nil {
+		return err
+	}
+
+	if err := r.Create(ctx, nm); err != nil {
+		klog.Errorf("%s --> Problems in NamespaceMap creation for the virtual node '%s'", err, n.GetName())
+		return err
+	}
+	klog.Infof(" Create the NamespaceMap '%s' for the virtual node '%s'", nm.GetName(), n.GetName())
+	return nil
+}
+
+// This function manages NamespaceMaps Lifecycle on the basis of NamespaceMaps' number.
+func (r *VirtualNodeReconciler) checkNamespaceMapPresence(ctx context.Context, n *corev1.Node) error {
```

Comment:
```suggestion
func (r *VirtualNodeReconciler) ensureNamespaceMapPresence(ctx context.Context, n *corev1.Node) error {
```

Same as before about name clarity.